app/workers/update_for_you_worker.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class UpdateForYouWorker
+  include Redisable
+  include Async
+  include Sidekiq::Worker
+
+  sidekiq_options retry: 0, queue: 'pull'
+
+  #  Fetch Acct Config from AcctRelay
+  #  Fetch Following from AcctRelay
+  #  Then status of those accounts from following locally
+  #  Finally send them to for_you_feed_worker
+  def perform(opts)
+    @personal = PersonalForYou.new
+    @acct = opts['acct']
+    @user = mammoth_user(@acct).wait
+    # This is temperary
+    @account = local_account
+
+    # Unable to resolve account
+    # Set Status to 'error'
+    if @account.nil?
+      update_user_status('error').wait
+      ResolveAccountWorker.perform_async(@acct)
+      return nil
+    end
+
+    # If rebuild is true, Zero Out User's for you feed
+    @personal.reset_feed(@account.id) if opts['rebuild']
+
+    push_status!
+
+    # Final Step:
+    # Set user's status to 'idle'
+    update_user_status('idle').wait
+  end
+
+  private
+
+  def push_status!
+    # Indirect Follow
+    push_indirect_following_status
+    # Direct Follows
+    push_following_status
+    # Public Feed
+  end
+
+  def update_user_status(status)
+    Async do
+      @personal.update_user(@acct, { status: status })
+    end
+  end
+
+  def local_account
+    domain = @user[:domain] == ENV['LOCAL_DOMAIN'] ? nil : @user[:domain]
+    Account.where(username: @user[:username], domain: domain).first
+  end
+
+  def mammoth_user(acct)
+    Async do
+      @personal.user(acct)
+    end
+  end
+
+  # TODO: update account.id to user.acct
+  # Return early if user setting is Zero, meaning 'off' from the iOS perspective
+  def push_following_status
+    user_setting = @user[:for_you_settings]
+
+    @personal.statuses_for_direct_follows(@acct)
+             .filter_map { |s| engagment_threshold(s, user_setting[:your_follows], 'following') }
+             .each { |s| ForYouFeedWorker.perform_async(s['id'], @account.id, 'personal') } unless user_setting[:your_follows].zero?
+  end
+
+  # Indirect Follows
+  def push_indirect_following_status
+    user_setting = @user[:for_you_settings]
+    return if user_setting[:friends_of_friends].zero?
```

Comment:
Not sure it matters, but… the code above uses the " unless ___" construct, and this code uses a "return if ___" construct. Maybe pick one style to use in both places. (?)

Reply:
oh. I had meant to change that back so they both had an early return. good call